### PR TITLE
fix(launchpad): Support migrating projects w/o pluginsFile 

### PR DESCRIPTION
**Pull request by lmiller1990**
Closes [UNIFY-1007](https://github.com/cypress-io/cypress/pull/19940)

Note: Targeting https://github.com/cypress-io/cypress/pull/19940

This handles the edge case of migrate a project with **no** `pluginsFile` (eg, the user deleted the default scaffolded `pluginsFile`). I found this when I tried migrating [this test project](https://github.com/cypress-io/cypress-test-nested-projects). Now it's working - we simply omit the `require('./cypress/plugins/index.js'`) file from the created `cypress.config.js`. 😎

### CODE REVIEW

File: packages/data-context/src/sources/migration/shouldShowSteps.ts [hunk 1]
```diff
@@ -96,6 +96,16 @@ export async function shouldShowRenameSupport (projectRoot: string, config: OldC
     return false
   }
 
+  // if they do not have a supportFile specified in cypress.json AND
+  // no default is present, skip this step.
+  if (
+    (config.e2e?.supportFile ?? undefined) === undefined &&
+    config.supportFile === undefined &&
+    !await tryGetDefaultLegacySupportFile(projectRoot)
```
[the remark this answers is not in the document]
Comment (AtofStryker) [reply packages/data-context/src/sources/migration/shouldShowSteps.ts:104]: Could probably do the same thing in `MigrationDataSource`

File: packages/data-context/src/sources/MigrationDataSource.ts [hunk 2]
```diff
@@ -39,28 +40,44 @@ export interface MigrationFile {
...
+  hasCustomIntegrationTestFiles: false,
+
+  hasCustomComponentFolder: false,
+  hasCustomComponentTestFiles: false,
+
+  hasCustomSupportFile: false,
+  hasComponentTesting: true,
+  hasE2ESpec: true,
+  hasPluginsFile: true,
+} as const
```
Comment (AtofStryker) [inline packages/data-context/src/sources/MigrationDataSource.ts:54]: Is the additional cast here needed?
```suggestion
}
```

File: packages/data-context/src/sources/MigrationDataSource.ts [hunk 3]
```diff
@@ -39,28 +40,44 @@ export interface MigrationFile {
...
+
 export class MigrationDataSource {
   private _config: OldCypressConfig | null = null
   private _step: MIGRATION_STEP = 'renameAuto'
   filteredSteps: MIGRATION_STEP[] = MIGRATION_STEPS.filter(() => true)
 
-  hasCustomIntegrationFolder: boolean = false
-  hasCustomIntegrationTestFiles: boolean = false
+  hasCustomIntegrationFolder: boolean = flags.hasCustomIntegrationFolder
```
Comment (AtofStryker) [inline packages/data-context/src/sources/MigrationDataSource.ts:61]: I think the type is inferred here, but I am guessing declaring the types here is intentionally explicit to make the type clear?

**Review by AtofStryker — APPROVED** [review]
🚀

File: packages/data-context/src/sources/MigrationDataSource.ts (hunk 2 above)
Comment (elevatebart) [reply packages/data-context/src/sources/MigrationDataSource.ts:51]: `as const` is needed here to help typescript making a type out of the keys.

see this blog post:
https://www.bscotch.net/post/typescript-as-const

File: packages/data-context/src/sources/MigrationDataSource.ts (hunk 3 above)
Comment (elevatebart) [reply packages/data-context/src/sources/MigrationDataSource.ts:58]: exactly

Comment (lmiller1990) [reply packages/data-context/src/sources/MigrationDataSource.ts:58]: yes!

**Pull request merged**